transports/noise/src/io/handshake.rs

```diff
@@ -167,12 +174,16 @@ pub(crate) async fn recv_empty<T>(state: &mut State<T>) -> Result<(), Error>
 where
     T: AsyncRead + Unpin,
 {
-    let msg = recv(state).await?;
-    if !msg.is_empty() {
-        return Err(
-            io::Error::new(io::ErrorKind::InvalidData, "Unexpected handshake payload.").into(),
-        );
+    let proto::NoiseHandshakePayload {
+        identity_key,
+        identity_sig,
+        extensions,
+    } = recv(state).await?;
+
+    if !(identity_key.is_empty() && identity_sig.is_empty() && extensions.is_none()) {
```

Comment:
We could use `MessageWrite::size` here and check that the struct is of length `0` when serialized. You reckon that is too much of a hack?

Reply:
I think either option works fine. I'll just go with `MessageWrite::size` since it's shorter